### PR TITLE
fix: move custom fields to JSON

### LINKED ISSUE #2: Refactor custom fields to use custom folder

### PR DESCRIPTION
**Pull request by Alchez**
Closes #2.

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -1,40 +1,10 @@
 name: Linters
 
 on:
-  push:
-    branches:
-      - main-v15
   pull_request:
 
 jobs:
-  conflicts:
```
Comment (agritheory) [inline .github/workflows/lint.yaml:10]: Doesn't hurt to run this check here

File: shipstation_integration/hooks.py [hunk 2]
```diff
@@ -61,8 +61,7 @@
 # Installation
 # ------------
 
-# TODO: move to custom JSON: https://github.com/agritheory/shipstation_integration/issues/2
-before_migrate = "shipstation_integration.tests.setup.setup_custom_fields"
+# before_migrate = "shipstation_integration.migrate.before_migrate"
```
Comment (agritheory) [inline shipstation_integration/hooks.py]: We need the after migrate hook here too to ensure these fields get sync'd. I would like to use the one in test_utils but it isn't ready yet

**Review by agritheory — CHANGES_REQUESTED** [review]
Add custom field pre-commits from test_utils

All custom fields and property setters should have a module of "Shipstation Integration"

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (Alchez) [reply .github/workflows/lint.yaml:10]: The pre-commit job below will run the same check, so this will be redundant

**Comment by Alchez** [comment]
@agritheory I've added the `validate_customizations` pre-commit hook, which also setup the module name in the JSON files.

File: shipstation_integration/hooks.py (hunk 2 above)
Comment (Alchez) [reply shipstation_integration/hooks.py]: Wouldn't the `bench migrate` command automatically sync up everything in the `custom` folder? Should this PR wait until the test-util hook is ready?

**Comment by agritheory** [comment]
@Alchez We can't run a pre-commit that is expecting a complete Frappe environment unless we build that environment.

**Comment by Alchez** [comment]
@agritheory makes sense. For this repo, I've skipped the `validate_customizations` in the default linter CI workflow, and instead added it to the Frappe CI workflow.

But the test is still failing and I had created a ticket over at https://github.com/agritheory/test_utils/issues/54 for the error. I think we'll have to update and document this further.

CC: @MyuddinKhatri

**Review by agritheory — APPROVED** [review]

**Pull request merged**